src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.cs

```diff
@@ -324,6 +324,11 @@ private void InitializeAbbreviatedEraNames(string localeName, CalendarId calenda
 
         internal static CalendarData GetCalendarData(CalendarId calendarId)
         {
+            if (GlobalizationMode.Invariant)
+            {
+                return CalendarData.Invariant;
+            }
```

Comment:
Do we already return invariant data for all calendarIds, and this is just doing the check earlier/differently to help the linker trim better?

Or is this actually changing behavior when GlobalizationMode.Invariant is set?

Reply:
In Invariant mode, we always use the Invariant culture when requesting any culture. so, I believe we are not changing any behavior for the invariant mode here.

Reply:
There is no behaviour change, PR only adds few explicit checks so less code is executed under GlobalizationMode.Invariant with the same result